common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -65,26 +65,39 @@ public static class Collation {
     public final BiFunction<UTF8String, UTF8String, Boolean> equalsFunction;
 
     /**
-     * Binary collation implies that UTF8Strings are considered equal only if they are
-     * byte for byte equal. All accent or case-insensitive collations are considered non-binary.
+     * Support for Binary Equality implies that it is possible to check equality on
+     * byte by byte level. This allows for the usage of binaryEquals call on UTF8Strings
+     * which is more performant than calls to external ICU library.
      */
-    public final boolean isBinaryCollation;
+    public final boolean supportsBinaryEquality;
+    /**
+     * Support for Binary Ordering implies that it is possible to check equality and compare on
```

Comment:
```suggestion
     * Support for Binary Ordering implies that it is possible to check ordering and compare on
```

Reply:
I wanted to say that this type of collation needs both equality and ordering to be possible. Should I maybe change the wording to equality and ordering.

Reply:
oh, then `... equality and ordering ...` sounds better.